### ISSUE TITLE
Node never finishes bootstrapping

### ISSUE THREAD

**Issue by danlaine** [ava-labs/gecko#209]
**Describe the bug**
Node doesn't finish bootstrapping

**To Reproduce**
Start a node

**Expected behavior**
Node finishes bootstrapping

**Screenshots**
See logs here: https://gist.github.com/moreati/91089756e85cc47d3755f7e3925b2368

**Operating System**
Ubuntu 20.04

**Additional Context**
* Go version 1.13.8 linux/amd64
* On commit 5846e69
* Running under systemd, and was installed by Ansible to /usr/local/bin

Reported by @moreati 

By submitting this issue I agree to the [Terms and Conditions](https://files.avalabs.org/legal/bug_hunt_q2_2020_terms.pdf) of the [Developer Accelerator Program](https://avalabs.org/dap).

**Comment by moreati** [comment]
This node was installed by the roles and playbook in https://github.com/ava-labs/gecko/pull/151

**Comment by moreati** [comment]
Previous Gecko versions installed by this playbook have synced, both on cascade and denali

**Comment by moreati** [comment]
I dropped back to Gecko 0.5.0 (built locally from git tag) and started again with a fresh database. Again, it did not sync. Of note
- 1 GB/1vCPU VM is clearly not enough for denali testnet. `ava` & `evm` regularly peg the CPU, kswapd0 is getting invoked a lot.
- ava process was killed several times during bootstrap (and restarted as per the system unit file)

Based on that I propose closing this issue as WONTFIX or similar

**Comment by youlaiwuqu** [comment]
1.On 06-03 about 11:42, I use commands as picture below to start running a node.
https://ibb.co/684776f
https://imgur.com/a/2O6XRYn
2.On 06-03 about 18:43, I find the node still don't finish sync as picture below. It's over 5 hours!
https://imgur.com/a/2O6XRYn
3.Then I joined discord, and follow the instructions to upgrade my node and run on 06-04 0:38 as below pictures.
https://imgur.com/a/t1kOKwY
https://imgur.com/a/cFo6uJw
4.On 06-04 0:50, I find full screen of "getfailed called without sending correspond........", I thought it were in a unhealthy state. I use CTRL+C and kill the node. See the picture below:
https://imgur.com/a/elBrguo
5.Does it matter the screen full of "getfailed called without sending correspond........."?
6.I restart my node following the instrucion as below picture on 06-04 about 02：41. On 06-04 about 08:51, still don't finish syc. It has been over 6 hours!
https://imgur.com/a/t1kOKwY
https://imgur.com/a/SGBziij
7.I use Opitical Network Unit(ONU), my network is fast and fine. But why I failed to connect to 127.0.0.1 port 9650 or 9651,9652......? Although I failed to connect to 127.0.0.1, why I have full screen of "getfailed called without sending correnspond.........."?
https://imgur.com/a/fGpDdJ8
https://imgur.com/a/EHTR73o
8.My email address is   ljjathena@163.com 
I'm on the waiting list. I run node on Ubuntu, windows' subsystem. Hoping I will become the validator. (It's too short from announcement to 1000 seats are full. Only about half of a day. ) Although I'm not a programmer, I have trigger many bugs for other blockchain projects(POW consensus) and get rewarded.

**Comment by moreati** [comment]
I've deployed (same Ansible playbook) a node to a 1 vCPU, 3.75 GB memory. To see if it copes with same vCPU, but more RAM

**Comment by youlaiwuqu** [comment]
9. For 7, I find I failed to connect to 127.0.0.1 port 9650 because I don't run a node and meanwhile use curl command. Now I use nohup ./ava & command and meanwhile use curl command, below picture shows my network is normal.(The programmer in go QQ group teach me.^_^)
https://imgur.com/a/x6i3sgN

**Comment by moreati** [comment]
> I've deployed (same Ansible playbook) a node to a 1 vCPU, 3.75 GB memory VM. To see if it copes with same vCPU, but more RAM

This node is not crash, but it's not finished bootstrapping, even after 18 hours. Log attached. It was run under systemd

```
alex@instance-1:~$ systemctl status ava | cat
● ava.service - AVA test node
     Loaded: loaded (/etc/systemd/system/ava.service; enabled; vendor preset: enabled)
     Active: active (running) since Thu 2020-06-04 07:10:34 UTC; 14h ago
       Docs: https://docs.ava.network/
   Main PID: 12547 (ava)
      Tasks: 21 (limit: 4410)
     Memory: 983.3M
     CGroup: /system.slice/ava.service
             ├─12547 /usr/local/bin/ava --public-ip=34.72.246.34 --http-host=localhost --db-dir=/var/local/lib/ava/db --plugin-dir=/usr/local/lib/ava/plugins --log-dir=/var/local/log/ava --log-level=info --staking-tls-cert-file=/var/local/lib/ava/staking/staker.crt --staking-tls-key-file=/var/local/lib/ava/staking/staker.key
             ├─13654 sh -c /usr/local/lib/ava/plugins/evm
             └─13655 /usr/local/lib/ava/plugins/evm
...
```
[instance-1.ava.log.gz](https://github.com/ava-labs/gecko/files/4732811/instance-1.ava.log.gz)

**Comment by drae12** [comment]
1. stop node 
2.  ./scripts/build.sh 
3. ./build/ava

**Comment by moreati** [comment]
Rebuilding produces a bit-for-bit identical binary

```
alex@instance-1:~/go/src/github.com/ava-labs/gecko$ shasum build/ava 
1a62ae5b51df89144fea4a1c82613865511e4b74  build/ava
alex@instance-1:~/go/src/github.com/ava-labs/gecko$ git pull
Already up to date.
alex@instance-1:~/go/src/github.com/ava-labs/gecko$ ./scripts/build.sh 
Downloading dependencies...
Building Gecko...
Building Coreth...
# gopkg.in/olebedev/go-duktape.v3
In file included from _cgo_export.c:4:
debugger.go:22:13: warning: ‘_duk_debugger_attach’ defined but not used [-Wunused-function]
Build Successful
alex@instance-1:~/go/src/github.com/ava-labs/gecko$ shasum build/ava 
1a62ae5b51df89144fea4a1c82613865511e4b74  build/ava
```

Were you thinking I was running an older binary?

**Comment by gin** [comment]
I had the same issue (bootstrapping stuck for 12hrs) with only port 9650 opened.
However, it was resolved after opening all ports, and restarting the client (Ctrl-C, then ./ava). Uncertain if it was because bootstrapping needed just a bit longer or if it was due to opening more ports. Probably just need more ports to be opened.

Observations:
In `main/params.go`:
* genesis seed nodes have port 21001
* bootstrap IP port in the comments have port 9630, 9631
* staking-port is 9651
* xput-server-port is 9652

----

**Operating System**
Ubuntu 20.04

**Additional Context**

Go version 1.14.4 linux/amd64
On commit 5846e69
Running via command line manually (with `ava`) and was installed with `scripts/build.sh` to `$GOPATH/.../ava-labs/gecko/build/`

**Comment by moreati** [comment]
0.5.3 has fixed this for me on the 1 vCPU, 3.75 GB memory VM. Still seeing how the underpowered DO droplet copes.

**Comment by moreati** [comment]
The underpowered DO droplet has also synced

**Comment by youlaiwuqu** [comment]
I have run node0.5.3, over 8 hours, but still don't finish  sync and get a 0 balance of X chain.

**Comment by danlaine** [comment]
Hi all. You may have more luck with the latest version, 0.5.4. 
Try that out, and comment here if you still face issues.
Thanks :)

**Comment by youlaiwuqu** [comment]
When to release the waiting list？ I'm on the waiting list but become.a.validator. Its unfair for us validators but on the waiting list. We contribute to the testnet but get nothing rewarded.
It's fair that the validator who finish challenges be rewarded not the ones who have fast hand speed to  regist.

**Comment by peterphan1996** [comment]
I'm on 0.5.5 and the issue above still exists

**Comment by danlaine** [comment]
Hi all, this should be resolved by #256
Closing this issue for now. If you still face this issue after upgrading, let me know and I will re-open this issue.

**Comment by romeshniriella** [comment]
Hello, I am facing this same issue avalanchego@v1.3.2 trying to run a flare node. 

I can see the following errors:

`~/flare/logs/chain/C$ cat 0.log`
ERROR[06-29|19:09:33.145] Failed to persist clean trie cache       error="cannot move temporary dir \"fastcache.tmp.794496106\" to \"\": rename fastcache.tmp.794496106 : no such file or directory"

`DEBUG[06-30|00:14:12] /home/flare/go/pkg/mod/github.com/ava-labs/avalanchego@v1.3.2/health/service.go#119: "isBootstrapped" returned unhealthy with: {"error":{"message":"primary subnet has not finished bootstrapping"},"timestamp":"2021-06-30T00:14:12.760932436Z","duration":3500,"contiguousFailures":2171,"timeOfFirstFailure":"2021-06-29T06:09:32.75930236Z"}`

`DEBUG[06-30|00:13:12] /home/flareadmin/go/pkg/mod/github.com/ava-labs/avalanchego@v1.3.2/health/service.go#119: "isBootstrapped" returned unhealthy with: {"error":{"message":"primary subnet has not finished bootstrapping"},"timestamp":"2021-06-30T00:13:12.760922719Z","duration":4300,"contiguousFailures":2169,"timeOfFirstFailure":"2021-06-29T06:09:32.75930236Z"}`

Does anyone know why this is happening?
